src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineUnspecTest.java

```diff
@@ -0,0 +1,187 @@
+            arguments(CaseDataBuilderUnspec.builder().atStateClaimSubmittedTwoRespondentRepresentatives().build()),
+            arguments(CaseDataBuilderUnspec.builder().atState1v2Solicitor1UnregisteredSolicitor2Registered().build())
+        );
+    }
+
+    @ParameterizedTest(name = "{index}: The state is transitioned correctly from DRAFT to CLAIM_SUBMITTED")
+    @MethodSource("caseDataStream")
+    void shouldReturnClaimSubmitted_whenCaseDataAtStateClaimSubmitted(CaseData caseData) {
+        // When
+        StateFlow stateFlow = stateFlowEngine.evaluate(caseData);
+
+        // Then Claim will go through state DRAFT and finish at state CLAIM_SUBMITTED
+        assertThat(stateFlow.getState())
+            .extracting(State::getName)
+            .isNotNull()
+            .isEqualTo(CLAIM_SUBMITTED.fullName());
+        assertThat(stateFlow.getStateHistory())
+            .hasSize(2)
+            .extracting(State::getName)
+            .containsExactly(
+                DRAFT.fullName(), CLAIM_SUBMITTED.fullName());
+    }
+
+    @ParameterizedTest(name = "{index}: The common state flow flags are present for a case transitioned to CLAIM_SUBMITTED")
+    @MethodSource("caseDataStream")
+    void shouldContainCommonFlags_whenCaseDataAtStateClaimSubmitted(CaseData caseData) {
+        //When
+        StateFlow stateFlow = stateFlowEngine.evaluate(caseData);
+
+        // Then Claim will have NOTICE_OF_CHANGE, GENERAL_APPLICATION_ENABLED, CERTIFICATE_OF_SERVICE and RPA_CONTINUOUS_FEED
+        assertThat(stateFlow.getFlags()).contains(
+            entry(FlowFlag.NOTICE_OF_CHANGE.name(), false),
+            entry(FlowFlag.GENERAL_APPLICATION_ENABLED.name(), false),
+            entry(FlowFlag.CERTIFICATE_OF_SERVICE.name(), false),
+            entry(FlowFlag.RPA_CONTINUOUS_FEED.name(), false)
+        );
+    }
+
+    @ParameterizedTest(name = "{index}: The state flow flag ONE_RESPONDENT_REPRESENTATIVE is set to true (for appropriate cases)")
+    @MethodSource("caseDataStreamOneRespondentRepresentative")
+    void shouldHaveOneRespondentRepresentativeFlagsSet_whenCaseDataAtStateClaimSubmitted(CaseData caseData) {
+        // When
+        StateFlow stateFlow = stateFlowEngine.evaluate(caseData);
+
+        // Then Claim will have ONE_RESPONDENT_REPRESENTATIVE set to true
+        assertThat(stateFlow.getFlags()).contains(
+            entry(FlowFlag.ONE_RESPONDENT_REPRESENTATIVE.name(), true)
+        );
+        assertThat(stateFlow.getFlags()).hasSize(6);    // bonus: if this fails, a flag was added/removed but tests were not updated
+    }
+
+    @ParameterizedTest(name = "{index}: The state flow flags ONE_RESPONDENT_REPRESENTATIVE " +
+        "and TWO_RESPONDENT_REPRESENTATIVES are present and set to the correct values (for appropriate cases)")
+    @MethodSource("caseDataStreamTwoRespondentRepresentatives")
+    void shouldReturnClaimSubmitted_whenCaseDataAtStateClaimSubmitted1v2DifferentSolicitor(CaseData caseData) {
```

Comment:
Can you update this name shouldReturnClaimSubmitted_whenCaseDataAtStateClaimSubmitted1v2DifferentSolicitor
to say - whenCaseDataAtStateClaimSubmitted1v2DifferentSolicitorBothRepresented

Reply:
Done